app/api/auth/oauth.py

```diff
@@ -0,0 +1,78 @@
+# app/auth/oauth.py
+
+from uuid import UUID
+from dotenv import load_dotenv
+from fastapi import Depends, HTTPException, status
+from fastapi.security import OAuth2PasswordBearer
+import os
+import jwt
+from app.models import TokenData
+from app.models import User
+from app.api.auth.auth import fake_db_user
+
+load_dotenv()
+
+oauth2_scheme = OAuth2PasswordBearer(tokenUrl="/api/v1/login/")
+
+async def get_current_user(token: str = Depends(oauth2_scheme)):
+    credentials_exception = HTTPException(
+        status_code=status.HTTP_401_UNAUTHORIZED,
+        detail="Could not validate credentials",
+        headers={"WWW-Authenticate": "Bearer"},
+    )
+    try:
+        payload = jwt.decode(token, os.getenv("SECRET_KEY"), algorithms=[os.getenv("ALGORITHM")])
+        user_id: str = payload.get("sub")
+        if user_id is None:
+            raise credentials_exception
+    except jwt.InvalidTokenError:
+        raise credentials_exception
+    user = next((u for u in fake_db_user.values() if str(u["id"]) == user_id), None) 
+    if user is None:
+        raise credentials_exception
+    return TokenData(user_id=UUID(user_id))
+
+async def get_current_active_user(current_user: TokenData = Depends(get_current_user)):
+    # Check if user_id is present
+    if current_user.user_id is None:
+        raise HTTPException(status_code=status.HTTP_401_UNAUTHORIZED, detail="Invalid or missing user_id")
+
+    # Find user by user_id in the fake database
+    user = next((u for u in fake_db_user.values() if str(u["id"]) == str(current_user.user_id)), None)
+    
+    # If user not found
+    if not user:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="User not found")
+    
+    # Check if the user is active
+    if not user["is_active"]:
+        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail="Inactive user")
+    
+    # Return the user as a dictionary (no need for .dict() since it's already a dict)
+    return user
+
+async def get_current_admin_user(current_user: User = Depends(get_current_active_user)):
+    try:
```

Comment:
the only check here should be if the user is admin, and we can know that from the user we return from get_current_user. If it's not an admin, return 403.

Reply:
why did you remove the get_current_user? I only said removing the get_current_active_user because that part should be in get_current_user

Reply:
I also said that get_current_user should return the user object so that we can use it in get_current_admin_user to check if it's an admin or not.

Reply:
```
def get_current_admin_user(current_user: User = Depends(get_current_user)) -> User:
    if not current_user.is_admin:
        raise HTTPException(
                status_code=status.HTTP_403_FORBIDDEN,
                detail="Admin privileges required"
            )
    return current_user
```